### PR TITLE
mariadb@10.5: update and upstream Linux fixes

### PR DESCRIPTION
**Pull request by danielnachun**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/mariadb@10.5.rb [hunk 1]
```diff
@@ -38,8 +38,8 @@ class MariadbAT105 < Formula
     # Need patch to remove MYSQL_SOURCE_DIR from include path because it contains
     # file called VERSION.
     # https://github.com/Homebrew/homebrew-core/pull/76887#issuecomment-840851149
-    # Reported upstream at https://jira.mariadb.org/browse/MDEV-7209 - this fix can be
-    # removed once that issue is closed and the fix has been merged into a stable release.
+    # Originally reported upstream at https://jira.mariadb.org/browse/MDEV-7209,
+    # but only partially fixed.
     patch :DATA
```
Comment (carlocab) [inline Formula/mariadb@10.5.rb:43]: Can we upstream this patch? It seems like upstream haven't fixed this particular issue yet.

Comment (danielnachun) [reply Formula/mariadb@10.5.rb:43]: Yeah I think they incorrectly marked it as resolved when it is not.  I'm not sure whether the fix we use would actually be acceptable for them, but it's worth submitting.  Fortunately the fix is extremely minor at this point.

Comment (danielnachun) [reply Formula/mariadb@10.5.rb:43]: I opened a PR to fix the `VERSION` issue for `mroonga` here: https://github.com/MariaDB/server/pull/1891.  Is it sufficient for to me to add this link as a comment for all versions of MariaDB?  If so then the MariaDB PRs should be ready to go.

Comment (carlocab) [reply Formula/mariadb@10.5.rb:43]: Yup, that should be fine, I think.

**Review by carlocab — APPROVED** [review]

File: Formula/mariadb@10.5.rb (hunk 1 above)
Comment (danielnachun) [reply Formula/mariadb@10.5.rb:43]: Great, I'll follow up with that in a syntax only PR so we can just get all these bottles published now.

**Pull request closed without merging**